cpu/esp32: use macros/units

fjmolinas opened the pull request:
### Contribution description

Use macros/unit instead of custom`MHZ` define.

### Testing procedure

- murdock should take of everything

### Issues/PRs references

benpicco commented on cpu/esp32/include/sdk_conf.h:
```diff
@@ -57,11 +57,11 @@ extern "C" {
-#ifndef CONFIG_ESP32_DEFAULT_CPU_FREQ_MHZ
-#define CONFIG_ESP32_DEFAULT_CPU_FREQ_MHZ       80
+#ifndef CLOCK_CORECLOCK
+#define CLOCK_CORECLOCK       80
```
`CLOCK_CORECLOCK` is usually in Hz, so this would be *more* inconsistent.

fjmolinas replied:
Right, I changed to its all HZ

gschorcht commented:
All these changes should be done very carefully, as `CONFIG_ESP32_DEFAULT_CPU_FREQ_MHZ` is a configuration variable that was not defined by me, but comes from ESP-IDF. Even more, I would prefer to leave it as it is, since it affects the vendor code. Changing it in the vendor code would make later ESP-IDF upgrades more difficult.

I could imagine to define `CLOCK_CORECLOCK` as a new variable and define it based on `CONFIG_ESP32_DEFAULT_CPU_FREQ_MHZ` or vice versa.

fjmolinas commented:
> I could imagine to define CLOCK_CORECLOCK as a new variable and define it based on CONFIG_ESP32_DEFAULT_CPU_FREQ_MHZ or vice versa.

That makes sense, will do!

fjmolinas commented:
> > I could imagine to define CLOCK_CORECLOCK as a new variable and define it based on CONFIG_ESP32_DEFAULT_CPU_FREQ_MHZ or vice versa.
> 
> That makes sense, will do!

I added just the `CLOCK_CORECLOCK` define and did not change to be used in code. I also switched to use the MHZ macro, but that could actually be split up if not liked.

jeandudey commented on cpu/esp8266/include/cpu_conf.h:
```diff
@@ -37,6 +42,8 @@ extern "C" {
 #ifndef ESP8266_CPU_FREQUENCY
 #define ESP8266_CPU_FREQUENCY   (80)
 #endif
+#define CLOCK_CORECLOCK         (MHZ(ESP8266_CPU_FREQUENCY))
```
```suggestion
```

jeandudey commented on cpu/esp32/include/sdk_conf.h:
```diff
@@ -22,18 +22,26 @@
 #ifndef CONFIG_ESP32_DEFAULT_CPU_FREQ_MHZ
 #define CONFIG_ESP32_DEFAULT_CPU_FREQ_MHZ   80
 #endif
+#define CLOCK_CORECLOCK                     (MHZ(CONFIG_ESP32_DEFAULT_CPU_FREQ_MHZ))
```
```suggestion
```

jeandudey commented:
>I also switched to use the MHZ macro, but that could actually be split up if not liked.

If the macro exists and we have the option to improve, so, why not, seems cleaner :-)

gschorcht commented:
@fjmolinas Let me come back to this PR.

`CLOCK_CORECLOCK` was added as define with PR #16361, OK. I also agree with changing `x * MHZ` to `MHZ(x)` in code belonging to RIOT port. But I strongly suggest to leave the vendor code in `cpu/esp32/vendor` as it is. Otherwise, upgrading to a newer version of the vendor code could become unnecessarily difficult.

gschorcht commented on cpu/esp32/vendor/esp-idf/soc/rtc_time.c:
```diff
@@ -65,7 +64,7 @@ static uint32_t rtc_clk_cal_internal(rtc_cal_sel_t cal_clk, uint32_t slowclk_cyc
         expected_freq = 150000; /* 150k internal oscillator */
     }
-    uint32_t us_time_estimate = (uint32_t) (((uint64_t) slowclk_cycles) * MHZ / expected_freq);
+    uint32_t us_time_estimate = (uint32_t) (MHZ(slowclk_cycles) / expected_freq);
```
I am not sure if `uint32_t` is sufficient when `slowclk_cycles` is multiplied by one million and then divided by `expected_freq`. I don't know what the maximum value of `slowclk_cycles` is.

gschorcht commented on cpu/esp32/vendor/esp-idf/esp32/clk.c:
```diff
@@ -44,15 +44,14 @@
 #include "soc/i2s_reg.h"
 #include "driver/periph_ctrl.h"
 #include "xtensa/core-macros.h"
+#include "macros/units.h"
```
Please don't change this file, its vendor code.

gschorcht commented on cpu/esp32/vendor/esp-idf/soc/rtc_time.c:
```diff
@@ -17,8 +17,7 @@
 #include "soc/timer_group_reg.h"
-
-#define MHZ (1000000)
+#include "macros/units.h"
```
Please don't change this file, its vendor code.

gschorcht reviewed (changes requested):
The title is confusing. This PR just replaces MHZ definitions by the `MHZ` macro from `marcos/unit.s.h`.

gschorcht commented:
Please rebase and update this PR.

aabadie commented:
Is this still relevant after #17342 and #17424 have been merged ?

gschorcht commented:
Please drop the changes in
```
cpu/esp32/include/sdk_conf.h
cpu/esp8266/include/cpu_conf.h
cpu/esp_common/freertos/task.c
```
becaue they are obsolete since already merged with other PRs.

Please drop the changes in
```
cpu/esp32/vendor/esp-idf/esp32/clk.c
cpu/esp32/vendor/esp-idf/soc/rtc_clk.c
cpu/esp32/vendor/esp-idf/soc/rtc_sleep.c
cpu/esp32/vendor/esp-idf/soc/rtc_time.c
```
because these files are vendor code.

Changes in
```
cpu/esp32/startup.c
cpu/esp32/syscalls.c
```
could be still provided by this PR.

gschorcht commented:
> Please drop the changes in
> 
> ```
> cpu/esp32/vendor/esp-idf/esp32/clk.c
> cpu/esp32/vendor/esp-idf/soc/rtc_clk.c
> cpu/esp32/vendor/esp-idf/soc/rtc_sleep.c
> cpu/esp32/vendor/esp-idf/soc/rtc_time.c
> ```
> 
> because these files are vendor code.

By the way, I am currently working on the ESP32 port for the latest SDK. If it is successful, all vendor code will be removed from the RIOT repository. This port is already working, but unfortunately I still have problems with `esp_wifi`. The station connects to the AP successfully, but disconnects again immediately afterwards and I don`t know why. I'm diving deep into the SDK in hopes of eventually figuring out the problem. No idea if it will succeed.

fjmolinas commented:
Changed PR title, left only those changes.

gschorcht commented on cpu/esp32/startup.c:
```diff
@@ -256,7 +256,7 @@ static void IRAM system_clk_init (void)
     }
 
-    uint32_t freq_before = rtc_clk_cpu_freq_value(rtc_clk_cpu_freq_get()) / MHZ;
+    uint32_t freq_before = rtc_clk_cpu_freq_value(rtc_clk_cpu_freq_get()) / MHZ(1) ;
```
```suggestion
    uint32_t freq_before = rtc_clk_cpu_freq_value(rtc_clk_cpu_freq_get()) / MHZ(1);
```

gschorcht reviewed (changes requested):
Please squash directly.
